plugins/bitbucket-cloud-common/scripts/generate-models.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Backstage Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+SCRIPT_DIR=$(dirname $0)
+PLUGIN_DIR="${SCRIPT_DIR}/.."
+
+"${SCRIPT_DIR}"/prepare-schema.js
+yarn --cwd "${PLUGIN_DIR}" openapi-generator-cli generate --generator-key backstage
```

Comment:
Does this always download the latest definition? I tried running the model locally and ended up with the following diff:

<details>
<summary>
Diff
</summary>

```diff
diff --git a/plugins/bitbucket-cloud-common/bitbucket-cloud.oas.json b/plugins/bitbucket-cloud-common/bitbucket-cloud.oas.json
index 7a226d365e..ce289aa67d 100644
--- a/plugins/bitbucket-cloud-common/bitbucket-cloud.oas.json
+++ b/plugins/bitbucket-cloud-common/bitbucket-cloud.oas.json
@@ -4467,7 +4467,16 @@
           {
             "name": "merge",
             "in": "query",
-            "description": "If true, the source commit is merged into the\ndestination commit, and then a diff from the\ndestination to the merge result is returned. If false,\na simple 'two dot' diff between the source and\ndestination is returned. True if omitted.",
+            "description": "This parameter is deprecated and will be removed at the end\nof 2022. The 'topic' parameter should be used instead. The\n'merge' and 'topic' parameters cannot be both used at the same\ntime.\n\nIf true, the source commit is merged into the\ndestination commit, and then a diff from the\ndestination to the merge result is returned. If false,\na simple 'two dot' diff between the source and\ndestination is returned. True if omitted.",
+            "required": false,
+            "schema": {
+              "type": "boolean"
+            }
+          },
+          {
+            "name": "topic",
+            "in": "query",
+            "description": "If true, returns 2-way 'three-dot' diff.\nThis is a diff between the source commit and the merge base\nof the source commit and the destination commit.\nIf false, a simple 'two dot' diff between the source and\ndestination is returned.",
             "required": false,
             "schema": {
               "type": "boolean"
@@ -4487,7 +4496,7 @@
             "api_key": []
           }
         ],
-        "description": "Produces a raw git-style diff.\n\n#### Single commit spec\n\nIf the `spec` argument to this API is a single commit, the diff is\nproduced against the first parent of the specified commit.\n\n#### Two commit spec\n\nTwo commits separated by `..` may be provided as the `spec`, e.g.,\n`3a8b42..9ff173`. When two commits are provided and the `merge` query\nparameter is true or absent, this API produces a 3-way diff, also\nreferred to as a merge diff. This is equivalent to merging the left\nbranch into the right branch and then computing the diff of the merge\ncommit against its first parent (the right branch). These diffs have\nthe same behavior as pull requests that show the 3-way diff, such as\nthe [Bitbucket Cloud Pull\nRequest](https://blog.developer.atlassian.com/a-better-pull-request/).\nFor a simple git-style diff, add `merge=false` to the query.\n\nThe two commits are interpreted as follows:\n\n* First commit: the commit containing the changes we wish to preview\n* Second commit: the commit representing the state to which we want to\n  compare the first commit\n* **Note**: This is the opposite of the order used in `git diff`.\n\n#### Comparison to patches\n\nWhile similar to patches, diffs:\n\n* Don't have a commit header (username, commit message, etc)\n* Support the optional `path=foo/bar.py` query param to filter\n  the diff to just that one file diff\n\n#### Response\n\nThe raw diff is returned as-is, in whatever encoding the files in the\nrepository use. It is not decoded into unicode. As such, the\ncontent-type is `text/plain`."
+        "description": "Produces a raw git-style diff.\n\n#### Single commit spec\n\nIf the `spec` argument to this API is a single commit, the diff is\nproduced against the first parent of the specified commit.\n\n#### Two commit spec\n\nTwo commits separated by `..` may be provided as the `spec`, e.g.,\n`3a8b42..9ff173`. When two commits are provided and the `topic` query\nparameter is true or absent, this API produces a 2-way three dot diff.\nThis is the diff between source commit and the merge base of the source\ncommit and the destination commit. When the `topic` query param is false,\na simple git-style diff is produced.\n\nThe two commits are interpreted as follows:\n\n* First commit: the commit containing the changes we wish to preview\n* Second commit: the commit representing the state to which we want to\n  compare the first commit\n* **Note**: This is the opposite of the order used in `git diff`.\n\n#### Comparison to patches\n\nWhile similar to patches, diffs:\n\n* Don't have a commit header (username, commit message, etc)\n* Support the optional `path=foo/bar.py` query param to filter\n  the diff to just that one file diff\n\n#### Response\n\nThe raw diff is returned as-is, in whatever encoding the files in the\nrepository use. It is not decoded into unicode. As such, the\ncontent-type is `text/plain`."
       },
       "parameters": [
         {
@@ -4556,7 +4565,7 @@
             "api_key": []
           }
         ],
-        "description": "Produces a response in JSON format with a record for every path\nmodified, including information on the type of the change and the\nnumber of lines added and removed.\n\n#### Single commit spec\n\nIf the `spec` argument to this API is a single commit, the diff is\nproduced against the first parent of the specified commit.\n\n#### Two commit spec\n\nTwo commits separated by `..` may be provided as the `spec`, e.g.,\n`3a8b42..9ff173`. When two commits are provided and the `merge` query\nparameter is true or absent, this API produces a 3-way diff, also\nreferred to as a merge diff. This is equivalent to merging the left\nbranch into the right branch and then computing the diff of the merge\ncommit against its first parent (the right branch). These diffs have\nthe same behavior as pull requests that show the 3-way diff, such as\nthe [Bitbucket Cloud Pull\nRequest](https://blog.developer.atlassian.com/a-better-pull-request/).\nFor a simple git-style diff, add `merge=false` to the query.\n\nThe two commits are interpreted as follows:\n\n* First commit: the commit containing the changes we wish to preview\n* Second commit: the commit representing the state to which we want to\n  compare the first commit\n* **Note**: This is the opposite of the order used in `git diff`.\n\n#### Sample output\n```\ncurl https://api.bitbucket.org/2.0/repositories/bitbucket/geordi/diffstat/d222fa2..e174964\n{\n    \"pagelen\": 500,\n    \"values\": [\n        {\n            \"type\": \"diffstat\",\n            \"status\": \"modified\",\n            \"lines_removed\": 1,\n            \"lines_added\": 2,\n            \"old\": {\n                \"path\": \"setup.py\",\n                \"escaped_path\": \"setup.py\",\n                \"type\": \"commit_file\",\n                \"links\": {\n                    \"self\": {\n                        \"href\": \"https://api.bitbucket.org/2.0/repositories/bitbucket/geordi/src/e1749643d655d7c7014001a6c0f58abaf42ad850/setup.py\"\n                    }\n                }\n            },\n            \"new\": {\n                \"path\": \"setup.py\",\n                \"escaped_path\": \"setup.py\",\n                \"type\": \"commit_file\",\n                \"links\": {\n                    \"self\": {\n                        \"href\": \"https://api.bitbucket.org/2.0/repositories/bitbucket/geordi/src/d222fa235229c55dad20b190b0b571adf737d5a6/setup.py\"\n                    }\n                }\n            }\n        }\n    ],\n    \"page\": 1,\n    \"size\": 1\n}\n```"
+        "description": "Produces a response in JSON format with a record for every path\nmodified, including information on the type of the change and the\nnumber of lines added and removed.\n\n#### Single commit spec\n\nIf the `spec` argument to this API is a single commit, the diff is\nproduced against the first parent of the specified commit.\n\n#### Two commit spec\n\nTwo commits separated by `..` may be provided as the `spec`, e.g.,\n`3a8b42..9ff173`. When two commits are provided and the `topic` query\nparameter is true or absent, this API produces a 2-way three dot diff.\nThis is the diff between source commit and the merge base of the source\ncommit and the destination commit. When the `topic` query param is false,\na simple git-style diff is produced.\n\nThe two commits are interpreted as follows:\n\n* First commit: the commit containing the changes we wish to preview\n* Second commit: the commit representing the state to which we want to\n  compare the first commit\n* **Note**: This is the opposite of the order used in `git diff`.\n\n#### Sample output\n```\ncurl https://api.bitbucket.org/2.0/repositories/bitbucket/geordi/diffstat/d222fa2..e174964\n{\n    \"pagelen\": 500,\n    \"values\": [\n        {\n            \"type\": \"diffstat\",\n            \"status\": \"modified\",\n            \"lines_removed\": 1,\n            \"lines_added\": 2,\n            \"old\": {\n                \"path\": \"setup.py\",\n                \"escaped_path\": \"setup.py\",\n                \"type\": \"commit_file\",\n                \"links\": {\n                    \"self\": {\n                        \"href\": \"https://api.bitbucket.org/2.0/repositories/bitbucket/geordi/src/e1749643d655d7c7014001a6c0f58abaf42ad850/setup.py\"\n                    }\n                }\n            },\n            \"new\": {\n                \"path\": \"setup.py\",\n                \"escaped_path\": \"setup.py\",\n                \"type\": \"commit_file\",\n                \"links\": {\n                    \"self\": {\n                        \"href\": \"https://api.bitbucket.org/2.0/repositories/bitbucket/geordi/src/d222fa235229c55dad20b190b0b571adf737d5a6/setup.py\"\n                    }\n                }\n            }\n        }\n    ],\n    \"page\": 1,\n    \"size\": 1\n}\n```"
       },
       "parameters": [
         {
@@ -4571,7 +4580,7 @@
         {
           "name": "merge",
           "in": "query",
-          "description": "If true, the source commit is merged into the\ndestination commit, and then a diffstat from the\ndestination to the merge result is returned. If false,\na simple 'two dot' diffstat between the source and\ndestination is returned. True if omitted.",
+          "description": "This parameter is deprecated and will be removed at the end\nof 2022. The 'topic' parameter should be used instead. The\n'merge' and 'topic' parameters cannot be both used at the same\ntime.\n\nIf true, the source commit is merged into the\ndestination commit, and then a diffstat from the\ndestination to the merge result is returned. If false,\na simple 'two dot' diffstat between the source and\ndestination is returned. True if omitted.",
           "required": false,
           "schema": {
             "type": "boolean"
@@ -4613,6 +4622,15 @@
             "type": "string"
           }
         },
+        {
+          "name": "topic",
+          "in": "query",
+          "description": "If true, returns 2-way 'three-dot' diff.\nThis is a diff between the source commit and the merge base\nof the source commit and the destination commit.\nIf false, a simple 'two dot' diff between the source and\ndestination is returned.",
+          "required": false,
+          "schema": {
+            "type": "boolean"
+          }
+        },
         {
           "name": "workspace",
           "in": "path",
@@ -7453,13 +7471,585 @@
             "content": {
               "application/json": {
                 "schema": {
-                  "$ref": "#/components/schemas/paginated_milestones"
+                  "$ref": "#/components/schemas/paginated_milestones"
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "The specified repository does not exist or does not have the issue tracker enabled.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          }
+        },
+        "tags": ["Issue tracker"],
+        "summary": "List milestones",
+        "security": [
+          {
+            "oauth2": ["issue"]
+          },
+          {
+            "basic": []
+          },
+          {
+            "api_key": []
+          }
+        ],
+        "description": "Returns the milestones that have been defined in the issue tracker.\n\nThis resource is only available on repositories that have the issue\ntracker enabled."
+      },
+      "parameters": [
+        {
+          "name": "repo_slug",
+          "in": "path",
+          "description": "This can either be the repository slug or the UUID of the repository,\nsurrounded by curly-braces, for example: `{repository UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        },
+        {
+          "name": "workspace",
+          "in": "path",
+          "description": "This can either be the workspace ID (slug) or the workspace UUID\nsurrounded by curly-braces, for example: `{workspace UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        }
+      ]
+    },
+    "/repositories/{workspace}/{repo_slug}/milestones/{milestone_id}": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "The specified milestone object.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/milestone"
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "The specified repository or milestone does not exist or does not have the issue tracker enabled.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          }
+        },
+        "tags": ["Issue tracker"],
+        "summary": "Get a milestone",
+        "security": [
+          {
+            "oauth2": ["issue"]
+          },
+          {
+            "basic": []
+          },
+          {
+            "api_key": []
+          }
+        ],
+        "description": "Returns the specified issue tracker milestone object."
+      },
+      "parameters": [
+        {
+          "name": "milestone_id",
+          "in": "path",
+          "description": "The milestone's id",
+          "required": true,
+          "schema": {
+            "type": "integer"
+          }
+        },
+        {
+          "name": "repo_slug",
+          "in": "path",
+          "description": "This can either be the repository slug or the UUID of the repository,\nsurrounded by curly-braces, for example: `{repository UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        },
+        {
+          "name": "workspace",
+          "in": "path",
+          "description": "This can either be the workspace ID (slug) or the workspace UUID\nsurrounded by curly-braces, for example: `{workspace UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        }
+      ]
+    },
+    "/repositories/{workspace}/{repo_slug}/patch/{spec}": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "The raw patches"
+          },
+          "555": {
+            "description": "If the diff was too large and timed out.\n\nSince this endpoint does not employ any form of pagination, but\ninstead returns the diff as a single document, it can run into\ntrouble on very large diffs. If Bitbucket times out in cases\nlike these, a 555 status code is returned.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          }
+        },
+        "tags": ["Commits"],
+        "summary": "Get a patch for two commits",
+        "security": [
+          {
+            "oauth2": ["repository"]
+          },
+          {
+            "basic": []
+          },
+          {
+            "api_key": []
+          }
+        ],
+        "description": "Produces a raw patch for a single commit (diffed against its first\nparent), or a patch-series for a revspec of 2 commits (e.g.\n`3a8b42..9ff173` where the first commit represents the source and the\nsecond commit the destination).\n\nIn case of the latter (diffing a revspec), a patch series is returned\nfor the commits on the source branch (`3a8b42` and its ancestors in\nour example).\n\nWhile similar to diffs, patches:\n\n* Have a commit header (username, commit message, etc)\n* Do not support the `path=foo/bar.py` query parameter\n\nThe raw patch is returned as-is, in whatever encoding the files in the\nrepository use. It is not decoded into unicode. As such, the\ncontent-type is `text/plain`."
+      },
+      "parameters": [
+        {
+          "name": "repo_slug",
+          "in": "path",
+          "description": "This can either be the repository slug or the UUID of the repository,\nsurrounded by curly-braces, for example: `{repository UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        },
+        {
+          "name": "spec",
+          "in": "path",
+          "description": "A commit SHA (e.g. `3a8b42`) or a commit range using double dot\nnotation (e.g. `3a8b42..9ff173`).\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        },
+        {
+          "name": "workspace",
+          "in": "path",
+          "description": "This can either be the workspace ID (slug) or the workspace UUID\nsurrounded by curly-braces, for example: `{workspace UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        }
+      ]
+    },
+    "/repositories/{workspace}/{repo_slug}/permissions-config/groups": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "Paginated of explicit group permissions on the repository.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/paginated_repository_group_permissions"
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "The user couldn't be authenticated.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "403": {
+            "description": "The requesting user isn't an admin of the repository.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "One or both of the workspace and repository doesn't exist for the given identifiers.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          }
+        },
+        "tags": ["Repositories"],
+        "summary": "List explicit group permissions for a repository",
+        "security": [
+          {
+            "oauth2": ["repository:admin"]
+          },
+          {
+            "basic": []
+          },
+          {
+            "api_key": []
+          }
+        ],
+        "description": "Returns a paginated list of explicit group permissions for the given repository.\nThis endpoint does not support BBQL features.\n\nExample:\n\n```\n$ curl https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/permissions-config/groups\n\nHTTP/1.1 200\nLocation: https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/permissions-config/groups\n\n{\n  \"pagelen\": 10,\n  \"values\": [\n    {\n      \"type\": \"repository_group_permission\",\n      \"group\": {\n        \"type\": \"group\",\n        \"name\": \"Administrators\",\n        \"slug\": \"administrators\"\n      },\n      \"permission\": \"admin\",\n      \"links\": {\n        \"self\": {\n          \"href\": \"https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/\n                   geordi/permissions-config/groups/administrators\"\n        }\n      }\n    },\n    {\n      \"type\": \"repository_group_permission\",\n      \"group\": {\n        \"type\": \"group\",\n        \"name\": \"Developers\",\n        \"slug\": \"developers\"\n      },\n      \"permission\": \"read\",\n      \"links\": {\n        \"self\": {\n          \"href\": \"https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/\n                   geordi/permissions-config/groups/developers\"\n        }\n      }\n    }\n  ],\n  \"page\": 1,\n  \"size\": 2\n}\n```"
+      },
+      "parameters": [
+        {
+          "name": "repo_slug",
+          "in": "path",
+          "description": "This can either be the repository slug or the UUID of the repository,\nsurrounded by curly-braces, for example: `{repository UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        },
+        {
+          "name": "workspace",
+          "in": "path",
+          "description": "This can either be the workspace ID (slug) or the workspace UUID\nsurrounded by curly-braces, for example: `{workspace UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        }
+      ]
+    },
+    "/repositories/{workspace}/{repo_slug}/permissions-config/groups/{group_slug}": {
+      "delete": {
+        "responses": {
+          "204": {
+            "description": "Group permission deleted"
+          },
+          "401": {
+            "description": "The user couldn't be authenticated.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "403": {
+            "description": "The requesting user isn't an admin of the repository, or the authentication method was not via app password.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "The workspace does not exist, the repository does not exist,or the group does not exist.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          }
+        },
+        "tags": ["Repositories"],
+        "summary": "Delete an explicit group permission for a repository",
+        "security": [
+          {
+            "oauth2": ["repository:admin"]
+          },
+          {
+            "basic": []
+          },
+          {
+            "api_key": []
+          }
+        ],
+        "description": "Deletes the repository group permission between the requested repository and group, if one exists.\n\nOnly users with admin permission for the repository may access this resource.\n\nExample:\n\n$ curl -X DELETE https://api.bitbucket.org/2.0/repositories/atlassian_tutorial\n/geordi/permissions-config/groups/developers\n\n\nHTTP/1.1 204"
+      },
+      "get": {
+        "responses": {
+          "200": {
+            "description": "Group permission for group slug and repository",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/repository_group_permission"
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "The user couldn't be authenticated.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "403": {
+            "description": "The requesting user isn't an admin of the repository.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "The given user, workspace, and/or repository could not be found",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          }
+        },
+        "tags": ["Repositories"],
+        "summary": "Get an explicit group permission for a repository",
+        "security": [
+          {
+            "oauth2": ["repository:admin"]
+          },
+          {
+            "basic": []
+          },
+          {
+            "api_key": []
+          }
+        ],
+        "description": "Returns the group permission for a given group slug and repository\n\nOnly users with admin permission for the repository may access this resource.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n* `none`\n\nExample:\n\n```\n$ curl https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/permissions-config/groups/developers\n\nHTTP/1.1 200\nLocation:\nhttps://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/permissions-config/groups/developers\n\n{\n    \"type\": \"repository_group_permission\",\n    \"group\": {\n        \"type\": \"group\",\n        \"name\": \"Developers\",\n        \"slug\": \"developers\"\n    },\n    \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"atlassian_tutorial/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n    },\n    \"permission\": \"read\",\n    \"links\": {\n      \"self\": {\n        \"href\":\n        \"https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/permissions-config/groups/developers\"\n      }\n    }\n}\n```"
+      },
+      "put": {
+        "responses": {
+          "200": {
+            "description": "Group permission updated",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/repository_group_permission"
+                }
+              }
+            }
+          },
+          "400": {
+            "description": "No permission value was provided or the value is invalid(not one of read, write, or admin)",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "The user couldn't be authenticated.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "403": {
+            "description": "The requesting user isn't an admin of the repository, or the authentication method was not via app password.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "The workspace does not exist, the repository does not exist,or the group does not exist.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          }
+        },
+        "tags": ["Repositories"],
+        "summary": "Update an explicit group permission for a repository",
+        "security": [
+          {
+            "oauth2": ["repository:admin"]
+          },
+          {
+            "basic": []
+          },
+          {
+            "api_key": []
+          }
+        ],
+        "description": "Updates the group permission if it exists.\n\nOnly users with admin permission for the repository may access this resource.\n\nThe only authentication method supported for this endpoint is via app passwords.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n\nExample:\n```\n$ curl -X PUT -H \"Content-Type: application/json\"\nhttps://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/permissions-config/groups/developers\n-d\n'{\n    \"permission\": \"write\"\n}'\n\nHTTP/1.1 200\nLocation:\nhttps://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/permissions-config/groups/developers\n\n{\n    \"type\": \"repository_group_permission\",\n    \"group\": {\n        \"type\": \"group\",\n        \"name\": \"Developers\",\n        \"slug\": \"developers\"\n    },\n    \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"atlassian_tutorial/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n    },\n    \"permission\": \"write\",\n    \"links\": {\n      \"self\": {\n        \"href\":\n        \"https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/permissions-config/groups/developers\"\n      }\n    }\n}\n```"
+      },
+      "parameters": [
+        {
+          "name": "group_slug",
+          "in": "path",
+          "description": "Slug of the requested group.",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        },
+        {
+          "name": "repo_slug",
+          "in": "path",
+          "description": "This can either be the repository slug or the UUID of the repository,\nsurrounded by curly-braces, for example: `{repository UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        },
+        {
+          "name": "workspace",
+          "in": "path",
+          "description": "This can either be the workspace ID (slug) or the workspace UUID\nsurrounded by curly-braces, for example: `{workspace UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        }
+      ]
+    },
+    "/repositories/{workspace}/{repo_slug}/permissions-config/users": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "Paginated of explicit user permissions on the repository.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/paginated_repository_user_permissions"
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "The user couldn't be authenticated.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "403": {
+            "description": "The requesting user isn't an admin of the repository.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "No repository exists for the given repo slug and workspace.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          }
+        },
+        "tags": ["Repositories"],
+        "summary": "List explicit user permissions for a repository",
+        "security": [
+          {
+            "oauth2": ["repository:admin"]
+          },
+          {
+            "basic": []
+          },
+          {
+            "api_key": []
+          }
+        ],
+        "description": "Returns a paginated list of explicit user permissions for the given repository.\n\nExample:\n\n```\n$ curl https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/permissions-config/users\n\n{\n  \"pagelen\": 10,\n  \"values\": [\n    {\n        \"type\": \"repository_user_permission\",\n        \"user\": {\n            \"type\": \"user\",\n            \"display_name\": \"Colin Cameron\",\n            \"uuid\": \"{d301aafa-d676-4ee0-88be-962be7417567}\",\n            \"account_id\": \"557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a\"\n        },\n        \"permission\": \"admin\",\n        \"links\": {\n          \"self\": {\n            \"href\": \"https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/\n                     permissions-config/users/557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a\"\n          }\n        }\n    },\n    {\n      \"type\": \"repository_user_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Sean Conaty\",\n        \"uuid\": \"{504c3b62-8120-4f0c-a7bc-87800b9d6f70}\",\n        \"account_id\": \"557058:ba8948b2-49da-43a9-9e8b-e7249b8e324c\"\n      },\n      \"permission\": \"write\",\n      \"links\": {\n        \"self\": {\n          \"href\": \"https://api.bitbucket.org/2.0//repositories/atlassian_tutorial/geordi/\n                   permissions-config/users/557058:ba8948b2-49da-43a9-9e8b-e7249b8e324c\"\n        }\n      }\n    }\n  ],\n  \"page\": 1,\n  \"size\": 2\n}\n```"
+      },
+      "parameters": [
+        {
+          "name": "repo_slug",
+          "in": "path",
+          "description": "This can either be the repository slug or the UUID of the repository,\nsurrounded by curly-braces, for example: `{repository UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        },
+        {
+          "name": "workspace",
+          "in": "path",
+          "description": "This can either be the workspace ID (slug) or the workspace UUID\nsurrounded by curly-braces, for example: `{workspace UUID}`.\n",
+          "required": true,
+          "schema": {
+            "type": "string"
+          }
+        }
+      ]
+    },
+    "/repositories/{workspace}/{repo_slug}/permissions-config/users/{selected_user_id}": {
+      "delete": {
+        "responses": {
+          "204": {
+            "description": "The repository user permission was deleted and no content returned."
+          },
+          "401": {
+            "description": "The user couldn't be authenticated.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "403": {
+            "description": "The requesting user isn't an admin of the repository, or the authentication method was not via app password.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
                 }
               }
             }
           },
           "404": {
-            "description": "The specified repository does not exist or does not have the issue tracker enabled.",
+            "description": "One or more of the workspace, repository, and user doesn't exist for the given identifiers.",
             "content": {
               "application/json": {
                 "schema": {
@@ -7469,11 +8059,11 @@
             }
           }
         },
-        "tags": ["Issue tracker"],
-        "summary": "List milestones",
+        "tags": ["Repositories"],
+        "summary": "Delete an explicit user permission for a repository",
         "security": [
           {
-            "oauth2": ["issue"]
+            "oauth2": ["repository:admin"]
           },
           {
             "basic": []
@@ -7482,44 +8072,42 @@
             "api_key": []
           }
         ],
-        "description": "Returns the milestones that have been defined in the issue tracker.\n\nThis resource is only available on repositories that have the issue\ntracker enabled."
+        "description": "Deletes the repository user permission between the requested repository and user, if one exists.\n\nOnly users with admin permission for the repository may access this resource.\n\nThe only authentication method for this endpoint is via app passwords.\n\n```\n$ curl -X DELETE https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/\npermissions-config/users/557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a\n\n\nHTTP/1.1 204\n```"
       },
-      "parameters": [
-        {
-          "name": "repo_slug",
-          "in": "path",
-          "description": "This can either be the repository slug or the UUID of the repository,\nsurrounded by curly-braces, for example: `{repository UUID}`.\n",
-          "required": true,
-          "schema": {
-            "type": "string"
-          }
-        },
-        {
-          "name": "workspace",
-          "in": "path",
-          "description": "This can either be the workspace ID (slug) or the workspace UUID\nsurrounded by curly-braces, for example: `{workspace UUID}`.\n",
-          "required": true,
-          "schema": {
-            "type": "string"
-          }
-        }
-      ]
-    },
-    "/repositories/{workspace}/{repo_slug}/milestones/{milestone_id}": {
       "get": {
         "responses": {
           "200": {
-            "description": "The specified milestone object.",
+            "description": "Explicit user permission for user and repository",
             "content": {
               "application/json": {
                 "schema": {
-                  "$ref": "#/components/schemas/milestone"
+                  "$ref": "#/components/schemas/repository_user_permission"
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "The user couldn't be authenticated.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "403": {
+            "description": "The requesting user isn't an admin of the repository.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
                 }
               }
             }
           },
           "404": {
-            "description": "The specified repository or milestone does not exist or does not have the issue tracker enabled.",
+            "description": "One or both of the workspace and repository doesn't exist for the given identifiers.",
             "content": {
               "application/json": {
                 "schema": {
@@ -7529,11 +8117,11 @@
             }
           }
         },
-        "tags": ["Issue tracker"],
-        "summary": "Get a milestone",
+        "tags": ["Repositories"],
+        "summary": "Get an explicit user permission for a repository",
         "security": [
           {
-            "oauth2": ["issue"]
+            "oauth2": ["repository:admin"]
           },
           {
             "basic": []
@@ -7542,46 +8130,52 @@
             "api_key": []
           }
         ],
-        "description": "Returns the specified issue tracker milestone object."
+        "description": "Returns the explicit user permission for a given user and repository.\n\nOnly users with admin permission for the repository may access this resource.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n* `none`\n\nExample:\n\n```\n$ curl 'https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/\n        permissions-config/users/557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a'\n\nHTTP/1.1 200\nLocation: 'https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/\n           permissions-config/users/557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a'\n\n{\n    \"type\": \"repository_user_permission\",\n    \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Colin Cameron\",\n        \"uuid\": \"{d301aafa-d676-4ee0-88be-962be7417567}\",\n        \"account_id\": \"557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a\"\n    },\n    \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"atlassian_tutorial/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n    },\n    \"permission\": \"admin\",\n    \"links\": {\n        \"self\": {\n            \"href\": \"https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/\n                     permissions-config/users/557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a\"\n        }\n    }\n}\n```"
       },
-      "parameters": [
-        {
-          "name": "milestone_id",
-          "in": "path",
-          "description": "The milestone's id",
-          "required": true,
-          "schema": {
-            "type": "integer"
-          }
-        },
-        {
-          "name": "repo_slug",
-          "in": "path",
-          "description": "This can either be the repository slug or the UUID of the repository,\nsurrounded by curly-braces, for example: `{repository UUID}`.\n",
-          "required": true,
-          "schema": {
-            "type": "string"
-          }
-        },
-        {
-          "name": "workspace",
-          "in": "path",
-          "description": "This can either be the workspace ID (slug) or the workspace UUID\nsurrounded by curly-braces, for example: `{workspace UUID}`.\n",
-          "required": true,
-          "schema": {
-            "type": "string"
-          }
-        }
-      ]
-    },
-    "/repositories/{workspace}/{repo_slug}/patch/{spec}": {
-      "get": {
+      "put": {
         "responses": {
           "200": {
-            "description": "The raw patches"
+            "description": "Explicit user permission updated",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/repository_user_permission"
+                }
+              }
+            }
           },
-          "555": {
-            "description": "If the diff was too large and timed out.\n\nSince this endpoint does not employ any form of pagination, but\ninstead returns the diff as a single document, it can run into\ntrouble on very large diffs. If Bitbucket times out in cases\nlike these, a 555 status code is returned.",
+          "400": {
+            "description": "No permission value was provided or the value is invalid (not one of read, write, or admin), or the selected user is not a valid user to update.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "The user couldn't be authenticated.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "403": {
+            "description": "The requesting user isn't an admin of the repository, or the authentication method was not via app password.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/error"
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "One or more of the workspace, repo, and selected user doesn't exist for the given identifiers.",
             "content": {
               "application/json": {
                 "schema": {
@@ -7591,11 +8185,11 @@
             }
           }
         },
-        "tags": ["Commits"],
-        "summary": "Get a patch for two commits",
+        "tags": ["Repositories"],
+        "summary": "Update an explicit user permission for a repository",
         "security": [
           {
-            "oauth2": ["repository"]
+            "oauth2": ["repository:admin"]
           },
           {
             "basic": []
@@ -7604,7 +8198,7 @@
             "api_key": []
           }
         ],
-        "description": "Produces a raw patch for a single commit (diffed against its first\nparent), or a patch-series for a revspec of 2 commits (e.g.\n`3a8b42..9ff173` where the first commit represents the source and the\nsecond commit the destination).\n\nIn case of the latter (diffing a revspec), a patch series is returned\nfor the commits on the source branch (`3a8b42` and its ancestors in\nour example).\n\nWhile similar to diffs, patches:\n\n* Have a commit header (username, commit message, etc)\n* Do not support the `path=foo/bar.py` query parameter\n\nThe raw patch is returned as-is, in whatever encoding the files in the\nrepository use. It is not decoded into unicode. As such, the\ncontent-type is `text/plain`."
+        "description": "Updates the explicit user permission for a given user and repository. The selected user must be a member of\nthe workspace, and cannot be the workspace owner.\nOnly users with admin permission for the repository may access this resource.\n\nThe only authentication method for this endpoint is via app passwords.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n\nExample:\n\n```\n$ curl -X PUT -H \"Content-Type: application/json\" 'https://api.bitbucket.org/2.0/repositories/\natlassian_tutorial/geordi/permissions-config/users/557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a'\n-d         '{\n    \"permission\": \"write\"\n}'\n\nHTTP/1.1 200\nLocation: 'https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/\npermissions-config/users/557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a'\n\n\n{\n    \"type\": \"repository_user_permission\",\n    \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Colin Cameron\",\n        \"uuid\": \"{d301aafa-d676-4ee0-88be-962be7417567}\",\n        \"account_id\": \"557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a\"\n    },\n    \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"atlassian_tutorial/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n    },\n    \"permission\": \"write\",\n    \"links\": {\n        \"self\": {\n            \"href\": \"https://api.bitbucket.org/2.0/repositories/atlassian_tutorial/geordi/\n                     permissions-config/users/557058:ba8948b2-49da-43a9-9e8b-e7249b8e324a\"\n        }\n    }\n}\n```"
       },
       "parameters": [
         {
@@ -7617,9 +8211,9 @@
           }
         },
         {
-          "name": "spec",
+          "name": "selected_user_id",
           "in": "path",
-          "description": "A commit SHA (e.g. `3a8b42`) or a commit range using double dot\nnotation (e.g. `3a8b42..9ff173`).\n",
+          "description": "This can either be the UUID of the account, surrounded by curly-braces, for\nexample: `{account UUID}`, OR an Atlassian Account ID.\n",
           "required": true,
           "schema": {
             "type": "string"
@@ -14887,7 +15481,7 @@
             "api_key": []
           }
         ],
-        "description": "Returns an object for each repository permission for all of a\nteam’s repositories.\n\n**This endpoint has been removed.\nYou should use the [workspace repository permissions](/cloud/bitbucket/rest/api-group-workspaces/#api-workspaces-workspace-permissions-repositories-get) endpoint instead.\nFor more information, see [the announcement](https://developer.atlassian.com/cloud/bitbucket/bitbucket-api-teams-deprecation/).**\n\nIf the username URL parameter refers to a user account instead of\na team account, an object containing the repository permissions\nof all the username's repositories will be returned.\n\nPermissions returned are effective permissions — the highest level of\npermission the user has. This does not include public repositories that\nusers are not granted any specific permission in, and does not\ndistinguish between direct and indirect privileges.\n\nOnly users with admin permission for the team may access this resource.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n\nExample:\n\n```\n$ curl https://api.bitbucket.org/2.0/teams/atlassian_tutorial/permissions/repositories\n\n{\n  \"pagelen\": 10,\n  \"values\": [\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Erik van Zijst\",\n        \"uuid\": \"{d301aafa-d676-4ee0-88be-962be7417567}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"admin\"\n    },\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Sean Conaty\",\n        \"uuid\": \"{504c3b62-8120-4f0c-a7bc-87800b9d6f70}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"write\"\n    }\n  ],\n  \"page\": 1,\n  \"size\": 2\n}\n```\n\nResults may be further [filtered or sorted](/cloud/bitbucket/rest/intro/#filtering)\nby repository, user, or permission by adding the following query string\nparameters:\n\n* `q=repository.name=\"geordi\"` or `q=permission>\"read\"`\n* `sort=user.display_name`\n\nNote that the query parameter values need to be URL escaped so that `=`\nwould become `%3D`.",
+        "description": "Returns an object for each repository permission for all of a\nteam’s repositories.\n\n**This endpoint has been removed.\nYou should use the [workspace repository permissions](/cloud/bitbucket/rest/api-group-workspaces/#api-workspaces-workspace-permissions-repositories-get) endpoint instead.\nFor more information, see [the announcement](https://developer.atlassian.com/cloud/bitbucket/bitbucket-api-teams-deprecation/).**\n\nIf the username URL parameter refers to a user account instead of\na team account, an object containing the repository permissions\nof all the username's repositories will be returned.\n\nPermissions returned are effective permissions — the highest level of\npermission the user has. This does not include public repositories that\nusers are not granted any specific permission in, and does not\ndistinguish between explicit and implicit privileges.\n\nOnly users with admin permission for the team may access this resource.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n\nExample:\n\n```\n$ curl https://api.bitbucket.org/2.0/teams/atlassian_tutorial/permissions/repositories\n\n{\n  \"pagelen\": 10,\n  \"values\": [\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Erik van Zijst\",\n        \"uuid\": \"{d301aafa-d676-4ee0-88be-962be7417567}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"admin\"\n    },\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Sean Conaty\",\n        \"uuid\": \"{504c3b62-8120-4f0c-a7bc-87800b9d6f70}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"write\"\n    }\n  ],\n  \"page\": 1,\n  \"size\": 2\n}\n```\n\nResults may be further [filtered or sorted](/cloud/bitbucket/rest/intro/#filtering)\nby repository, user, or permission by adding the following query string\nparameters:\n\n* `q=repository.name=\"geordi\"` or `q=permission>\"read\"`\n* `sort=user.display_name`\n\nNote that the query parameter values need to be URL escaped so that `=`\nwould become `%3D`.",
         "deprecated": true
       },
       "parameters": [
@@ -14959,7 +15553,7 @@
             "api_key": []
           }
         ],
-        "description": "Returns an object for each repository permission of a given repository.\n\n**This endpoint has been removed.\nYou should use the [workspace repository permissions](/cloud/bitbucket/rest/api-group-workspaces/#api-workspaces-workspace-permissions-repositories-repo-slug-get) endpoint instead.\nFor more information, see [the announcement](https://developer.atlassian.com/cloud/bitbucket/bitbucket-api-teams-deprecation/).**\n\nIf the username URL parameter refers to a user account instead of\na team account, an object containing the repository permissions\nof the username's repository will be returned.\n\nPermissions returned are effective permissions — the highest level of\npermission the user has. This does not include public repositories that\nusers are not granted any specific permission in, and does not\ndistinguish between direct and indirect privileges.\n\nOnly users with admin permission for the repository may access this resource.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n\nExample:\n\n```\n$ curl https://api.bitbucket.org/2.0/teams/atlassian_tutorial/permissions/repositories/geordi\n\n{\n  \"pagelen\": 10,\n  \"values\": [\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Erik van Zijst\",\n        \"uuid\": \"{d301aafa-d676-4ee0-88be-962be7417567}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"admin\"\n    },\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Sean Conaty\",\n        \"uuid\": \"{504c3b62-8120-4f0c-a7bc-87800b9d6f70}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"write\"\n    }\n  ],\n  \"page\": 1,\n  \"size\": 2\n}\n```\n\nResults may be further [filtered or sorted](/cloud/bitbucket/rest/intro/#filtering)\nby user, or permission by adding the following query string parameters:\n\n* `q=permission>\"read\"`\n* `sort=user.display_name`\n\nNote that the query parameter values need to be URL escaped so that `=`\nwould become `%3D`.",
+        "description": "Returns an object for each repository permission of a given repository.\n\n**This endpoint has been removed.\nYou should use the [workspace repository permissions](/cloud/bitbucket/rest/api-group-workspaces/#api-workspaces-workspace-permissions-repositories-repo-slug-get) endpoint instead.\nFor more information, see [the announcement](https://developer.atlassian.com/cloud/bitbucket/bitbucket-api-teams-deprecation/).**\n\nIf the username URL parameter refers to a user account instead of\na team account, an object containing the repository permissions\nof the username's repository will be returned.\n\nPermissions returned are effective permissions — the highest level of\npermission the user has. This does not include public repositories that\nusers are not granted any specific permission in, and does not\ndistinguish between explicit and implicit privileges.\n\nOnly users with admin permission for the repository may access this resource.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n\nExample:\n\n```\n$ curl https://api.bitbucket.org/2.0/teams/atlassian_tutorial/permissions/repositories/geordi\n\n{\n  \"pagelen\": 10,\n  \"values\": [\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Erik van Zijst\",\n        \"uuid\": \"{d301aafa-d676-4ee0-88be-962be7417567}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"admin\"\n    },\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"display_name\": \"Sean Conaty\",\n        \"uuid\": \"{504c3b62-8120-4f0c-a7bc-87800b9d6f70}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"write\"\n    }\n  ],\n  \"page\": 1,\n  \"size\": 2\n}\n```\n\nResults may be further [filtered or sorted](/cloud/bitbucket/rest/intro/#filtering)\nby user, or permission by adding the following query string parameters:\n\n* `q=permission>\"read\"`\n* `sort=user.display_name`\n\nNote that the query parameter values need to be URL escaped so that `=`\nwould become `%3D`.",
         "deprecated": true
       },
       "parameters": [
@@ -15828,7 +16422,7 @@
             "api_key": []
           }
         ],
-        "description": "Returns an object for each repository the caller has explicit access\nto and their effective permission — the highest level of permission the\ncaller has. This does not return public repositories that the user was\nnot granted any specific permission in, and does not distinguish between\ndirect and indirect privileges.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n\nExample:\n\n```\n$ curl https://api.bitbucket.org/2.0/user/permissions/repositories\n\n{\n  \"pagelen\": 10,\n  \"values\": [\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"nickname\": \"evzijst\",\n        \"display_name\": \"Erik van Zijst\",\n        \"uuid\": \"{d301aafa-d676-4ee0-88be-962be7417567}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"admin\"\n    }\n  ],\n  \"page\": 1,\n  \"size\": 1\n}\n```\n\nResults may be further [filtered or sorted](/cloud/bitbucket/rest/intro/#filtering) by\nrepository or permission by adding the following query string\nparameters:\n\n* `q=repository.name=\"geordi\"` or `q=permission>\"read\"`\n* `sort=repository.name`\n\nNote that the query parameter values need to be URL escaped so that `=`\nwould become `%3D`."
+        "description": "Returns an object for each repository the caller has explicit access\nto and their effective permission — the highest level of permission the\ncaller has. This does not return public repositories that the user was\nnot granted any specific permission in, and does not distinguish between\nexplicit and implicit privileges.\n\nPermissions can be:\n\n* `admin`\n* `write`\n* `read`\n\nExample:\n\n```\n$ curl https://api.bitbucket.org/2.0/user/permissions/repositories\n\n{\n  \"pagelen\": 10,\n  \"values\": [\n    {\n      \"type\": \"repository_permission\",\n      \"user\": {\n        \"type\": \"user\",\n        \"nickname\": \"evzijst\",\n        \"display_name\": \"Erik van Zijst\",\n        \"uuid\": \"{d301aafa-d676-4ee0-88be-962be7417567}\"\n      },\n      \"repository\": {\n        \"type\": \"repository\",\n        \"name\": \"geordi\",\n        \"full_name\": \"bitbucket/geordi\",\n        \"uuid\": \"{85d08b4e-571d-44e9-a507-fa476535aa98}\"\n      },\n      \"permission\": \"admin\"\n    }\n  ],\n  \"page\": 1,\n  \"size\": 1\n}\n```\n\nResults may be further [filtered or sorted](/cloud/bitbucket/rest/intro/#filtering) by\nrepository or permission by adding the following query string\nparameters:\n\n* `q=repository.name=\"geordi\"` or `q=permission>\"read\"`\n* `sort=repository.name`\n\nNote that the query parameter values need to be URL escaped so that `=`\nwould become `%3D`."
       },
       "parameters": []
     },
@@ -18349,7 +18943,7 @@
       "description": "A workspace is where you create repositories, collaborate on\nyour code, and organize different streams of work in your Bitbucket\nCloud account. Workspaces replace the use of teams and users in API\ncalls.\n"
     }
   ],
-  "x-revision": "f5ed47932b76",
+  "x-revision": "46fdb03c2e7c",
   "x-atlassian-narrative": {
     "documents": [
       {
@@ -21486,6 +22080,29 @@
           }
         ]
       },
+      "paginated_repository_group_permissions": {
+        "title": "Paginated Repository Group Permissions",
+        "description": "A paginated list of repository group permissions.",
+        "allOf": [
+          {
+            "$ref": "#/components/schemas/paginated"
+          },
+          {
+            "type": "object",
+            "properties": {
+              "values": {
+                "type": "array",
+                "items": {
+                  "$ref": "#/components/schemas/repository_group_permission"
+                },
+                "minItems": 0,
+                "uniqueItems": true,
+                "description": "The values of the current page."
+              }
+            }
+          }
+        ]
+      },
       "paginated_repository_permissions": {
         "title": "Paginated Repository Permissions",
         "description": "A paginated list of repository permissions.",
@@ -21509,6 +22126,29 @@
           }
         ]
       },
+      "paginated_repository_user_permissions": {
+        "title": "Paginated Repository User Permissions",
+        "description": "A paginated list of repository user permissions.",
+        "allOf": [
+          {
+            "$ref": "#/components/schemas/paginated"
+          },
+          {
+            "type": "object",
+            "properties": {
+              "values": {
+                "type": "array",
+                "items": {
+                  "$ref": "#/components/schemas/repository_user_permission"
+                },
+                "minItems": 0,
+                "uniqueItems": true,
+                "description": "The values of the current page."
+              }
+            }
+          }
+        ]
+      },
       "paginated_snippet_comments": {
         "title": "Paginated Snippet Comments",
         "description": "A paginated list of snippet comments.",
@@ -23943,6 +24583,49 @@
         "title": "Repository",
         "description": "A Bitbucket repository."
       },
+      "repository_group_permission": {
+        "type": "object",
+        "title": "Repository Group Permission",
+        "description": "A group's permission for a given repository.",
+        "properties": {
+          "group": {
+            "$ref": "#/components/schemas/group"
+          },
+          "links": {
+            "type": "object",
+            "properties": {
+              "self": {
+                "type": "object",
+                "title": "Link",
+                "description": "A link to a resource related to this object.",
+                "properties": {
+                  "href": {
+                    "type": "string",
+                    "format": "uri"
+                  },
+                  "name": {
+                    "type": "string"
+                  }
+                },
+                "additionalProperties": false
+              }
+            },
+            "additionalProperties": false
+          },
+          "permission": {
+            "type": "string",
+            "enum": ["admin", "write", "read", "none"]
+          },
+          "repository": {
+            "$ref": "#/components/schemas/repository"
+          },
+          "type": {
+            "type": "string"
+          }
+        },
+        "required": ["type"],
+        "additionalProperties": true
+      },
       "repository_permission": {
         "type": "object",
         "title": "Repository Permission",
@@ -23965,6 +24648,49 @@
         "required": ["type"],
         "additionalProperties": true
       },
+      "repository_user_permission": {
+        "type": "object",
+        "title": "Repository User Permission",
+        "description": "A user's direct permission for a given repository.",
+        "properties": {
+          "links": {
+            "type": "object",
+            "properties": {
+              "self": {
+                "type": "object",
+                "title": "Link",
+                "description": "A link to a resource related to this object.",
+                "properties": {
+                  "href": {
+                    "type": "string",
+                    "format": "uri"
+                  },
+                  "name": {
+                    "type": "string"
+                  }
+                },
+                "additionalProperties": false
+              }
+            },
+            "additionalProperties": false
+          },
+          "permission": {
+            "type": "string",
+            "enum": ["admin", "write", "read", "none"]
+          },
+          "repository": {
+            "$ref": "#/components/schemas/repository"
+          },
+          "type": {
+            "type": "string"
+          },
+          "user": {
+            "$ref": "#/components/schemas/user"
+          }
+        },
+        "required": ["type"],
+        "additionalProperties": true
+      },
       "search_code_search_result": {
         "type": "object",
         "properties": {
diff --git a/plugins/bitbucket-cloud-common/src/models/index.ts b/plugins/bitbucket-cloud-common/src/models/index.ts
index 655e14d0e9..197f0b8109 100644
--- a/plugins/bitbucket-cloud-common/src/models/index.ts
+++ b/plugins/bitbucket-cloud-common/src/models/index.ts
@@ -1801,6 +1801,18 @@ export namespace Models {
     values?: Set<Repository>;
   }
 
+  /**
+   * A paginated list of repository group permissions.
+   * @public
+   */
+  export interface PaginatedRepositoryGroupPermissions
+    extends Paginated<RepositoryGroupPermission> {
+    /**
+     * The values of the current page.
+     */
+    values?: Set<RepositoryGroupPermission>;
+  }
+
   /**
    * A paginated list of repository permissions.
    * @public
@@ -1813,6 +1825,18 @@ export namespace Models {
     values?: Set<RepositoryPermission>;
   }
 
+  /**
+   * A paginated list of repository user permissions.
+   * @public
+   */
+  export interface PaginatedRepositoryUserPermissions
+    extends Paginated<RepositoryUserPermission> {
+    /**
+     * The values of the current page.
+     */
+    values?: Set<RepositoryUserPermission>;
+  }
+
   /**
    * A paginated list of snippet comments.
    * @public
@@ -3616,6 +3640,35 @@ export namespace Models {
     watchers?: Link;
   }
 
+  /**
+   * A group's permission for a given repository.
+   * @public
+   */
+  export interface RepositoryGroupPermission {
+    [key: string]: unknown;
+    group?: Group;
+    links?: BranchingModelSettingsLinks;
+    permission?: RepositoryGroupPermissionPermissionEnum;
+    repository?: Repository;
+    type: string;
+  }
+
+  /**
+   * @public
+   */
+  export const RepositoryGroupPermissionPermissionEnum = {
+    Admin: 'admin',
+    Write: 'write',
+    Read: 'read',
+    None: 'none',
+  } as const;
+
+  /**
+   * @public
+   */
+  export type RepositoryGroupPermissionPermissionEnum =
+    typeof RepositoryGroupPermissionPermissionEnum[keyof typeof RepositoryGroupPermissionPermissionEnum];
+
   /**
    * A user's permission for a given repository.
    * @public
@@ -3644,6 +3697,35 @@ export namespace Models {
   export type RepositoryPermissionPermissionEnum =
     typeof RepositoryPermissionPermissionEnum[keyof typeof RepositoryPermissionPermissionEnum];
 
+  /**
+   * A user's direct permission for a given repository.
+   * @public
+   */
+  export interface RepositoryUserPermission {
+    [key: string]: unknown;
+    links?: BranchingModelSettingsLinks;
+    permission?: RepositoryUserPermissionPermissionEnum;
+    repository?: Repository;
+    type: string;
+    user?: User;
+  }
+
+  /**
+   * @public
+   */
+  export const RepositoryUserPermissionPermissionEnum = {
+    Admin: 'admin',
+    Write: 'write',
+    Read: 'read',
+    None: 'none',
+  } as const;
+
+  /**
+   * @public
+   */
+  export type RepositoryUserPermissionPermissionEnum =
+    typeof RepositoryUserPermissionPermissionEnum[keyof typeof RepositoryUserPermissionPermissionEnum];
+
   /**
    * @public
    */
```
</details>

Feeling it's very important that this generation is stable across systems and idempotent. If ^ is the case, is it possible to split it apart so that updating the model is a separate action from downloading the latest definitions?

Reply:
Yes, it downloads the latest spec. I guess usually you want to rerun it if the external spec changed.

I can split it up.

Reply:
and seems like I need to rerun it :D